test/unit/Reflection/ReflectionParameterTest.php

```diff
@@ -571,18 +575,70 @@ public function testGetStartColumnAndEndColumn(string $php, int $startColumn, in
     {
-        $php = '<?php function foo($boo) {}';
+        $reflector          = $this->createMock(Reflector::class);
+        $parameterNode      = new Node\Param(new Node\Expr\Variable('foo'));
+        $functionReflection = $this->createMock(ReflectionFunction::class);
 
-        $reflector = new DefaultReflector(new StringSourceLocator($php, $this->astLocator));
-        $function  = $reflector->reflectFunction('foo');
-        $parameter = $function->getParameter('boo');
+        $parameterReflection = ReflectionParameter::createFromNode(
+            $reflector,
+            $parameterNode,
+            $functionReflection,
+            0,
+            false,
+        );
+
+        self::expectException(RuntimeException::class);
+        $parameterReflection->getStartColumn();
+    }
+
+    public function testGetEndColumnThrowsExceptionWhenMissing(): void
+    {
+        $reflector          = $this->createMock(Reflector::class);
+        $parameterNode      = new Node\Param(new Node\Expr\Variable('foo'));
+        $functionReflection = $this->createMock(ReflectionFunction::class);
+
+        $parameterReflection = ReflectionParameter::createFromNode(
+            $reflector,
+            $parameterNode,
+            $functionReflection,
+            0,
+            false,
+        );
+
+        self::expectException(RuntimeException::class);
+        $parameterReflection->getEndColumn();
+    }
+
+    public function testGetStartColumnThrowsExceptionForMagicallyAddedEnumMethod(): void
+    {
+        $reflector = new DefaultReflector(new AggregateSourceLocator([
+            new SingleFileSourceLocator(__DIR__ . '/../Fixture/Enums.php', $this->astLocator),
+            BetterReflectionSingleton::instance()->sourceLocator(),
+        ]));
+
+        $classReflection     = $reflector->reflectClass(StringEnum::class);
+        $methodReflection    = $classReflection->getMethod('tryFrom');
+        $parameterReflection = $methodReflection->getParameter('value');
+
+        self::expectException(RuntimeException::class);
+        $parameterReflection->getStartColumn();
+    }
+
+    public function testGetEndColumnThrowsExceptionForMagicallyAddedEnumMethod(): void
+    {
+        $reflector = new DefaultReflector(new AggregateSourceLocator([
+            new SingleFileSourceLocator(__DIR__ . '/../Fixture/Enums.php', $this->astLocator),
+            BetterReflectionSingleton::instance()->sourceLocator(),
+        ]));
 
-        $ast = $parameter->getAst();
+        $classReflection     = $reflector->reflectClass(StringEnum::class);
+        $methodReflection    = $classReflection->getMethod('tryFrom');
+        $parameterReflection = $methodReflection->getParameter('value');
 
-        self::assertInstanceOf(Param::class, $ast);
-        self::assertSame('boo', $ast->var->name);
+        self::expectException(RuntimeException::class);
+        $parameterReflection->getEndColumn();
```

Comment:
I think this is an acceptable behavior for now: the `RuntimeException` is not super nice, but perhaps we can document that `getEndColumn()` throws? 🤔